### PR TITLE
APP-1629: Create users

### PR DESCRIPTION
**Pull request by crepache**
- Create agents, add on queue room and add on the test context
- Create test context.

### CODE REVIEW

File: helpdesk-bot/src/test/java/org/symphonyoss/symphony/bots/helpdesk/bot/it/TestContext.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Created by crepache on 22/02/18.
+ * Class to store the variables to use on any part of tests.
+ */
+public class TestContext {
+
+  private static TestContext instance;
```
Comment (robson-sanchez) [inline helpdesk-bot/src/test/java/org/symphonyoss/symphony/bots/helpdesk/bot/it/TestContext.java]: You should perform eager initialization and then remove the synchronized directive from the getInstance method.

```
private static final TestContext INSTANCE = new TestContext();
```

File: helpdesk-bot/src/test/java/org/symphonyoss/symphony/bots/helpdesk/bot/it/TestContext.java [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+  private Map<UsersEnum, SymUser> users = new HashMap();
+
+  private Room queueRoom;
+
+  private TestContext() {
+  }
+
+  public static synchronized TestContext getInstance() {
```
Comment (robson-sanchez) [inline helpdesk-bot/src/test/java/org/symphonyoss/symphony/bots/helpdesk/bot/it/TestContext.java]: You might remove this synchronized directive if you perform the eager initialization. You should remove the null check as well.

File: helpdesk-bot/src/test/java/org/symphonyoss/symphony/bots/helpdesk/bot/it/TestContext.java [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+  public SymUser getUser(UsersEnum key) {
+    if (key == null) {
+      return null;
+    }
+
+    return users.get(key);
+  }
+
+  public void setUsers(UsersEnum key, SymUser user) {
+    users.put(key, user);
```
Comment (robson-sanchez) [inline helpdesk-bot/src/test/java/org/symphonyoss/symphony/bots/helpdesk/bot/it/TestContext.java]: Please verify the null key and throws IllegalArgumentException if required.

**Review by robson-sanchez — CHANGES_REQUESTED** [review]

File: helpdesk-bot/src/test/java/org/symphonyoss/symphony/bots/helpdesk/bot/it/TestContext.java (hunk 1 above)
Comment (crepache) [reply helpdesk-bot/src/test/java/org/symphonyoss/symphony/bots/helpdesk/bot/it/TestContext.java]: Done.

File: helpdesk-bot/src/test/java/org/symphonyoss/symphony/bots/helpdesk/bot/it/TestContext.java (hunk 2 above)
Comment (crepache) [reply helpdesk-bot/src/test/java/org/symphonyoss/symphony/bots/helpdesk/bot/it/TestContext.java]: Done.

File: helpdesk-bot/src/test/java/org/symphonyoss/symphony/bots/helpdesk/bot/it/TestContext.java (hunk 3 above)
Comment (crepache) [reply helpdesk-bot/src/test/java/org/symphonyoss/symphony/bots/helpdesk/bot/it/TestContext.java]: Done.

**Review by robson-sanchez — APPROVED** [review]

**Pull request merged**